hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -78,7 +78,7 @@ public static Option<HoodieCommitMetadata> resolveWriteConflictIfAny(
       final ConcurrentOperation thisOperation = new ConcurrentOperation(currentTxnOwnerInstant.get(), thisCommitMetadata.orElse(new HoodieCommitMetadata()));
       instantStream.forEach(instant -> {
         try {
-          ConcurrentOperation otherOperation = new ConcurrentOperation(instant, table.getMetaClient());
+          ConcurrentOperation otherOperation = new ConcurrentOperation(instant, TimelineUtils.getCommitMetadata(instant, table.getActiveTimeline()));
           if (resolutionStrategy.hasConflict(thisOperation, otherOperation)) {
```

Comment:
Need to figure out why the original commit wrap the commit metadaat using `HoodieArchivedMetaEntry`, @n3nash can you give some clarification here?

Reply:
I think I find the reason. The instant may be compaction.request, in this case we can't use TimelineUtils.getCommitMetadata.